pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -376,6 +378,18 @@ func (r *ReferenceAdapter) EnsureProjectCleanedUp() error {
 		if err != nil {
 			return err
 		}
+	} else {
```

Comment:
I think we need to analyses the logic here. I am not sure this else is necessary.